packages/pouchdb-core/src/parseAdapter.js

```diff
@@ -0,0 +1,49 @@
+import PouchDB from './constructor';
+import { guardedConsole, hasLocalStorage } from 'pouchdb-utils';
+
+function parseAdapter(name, opts) {
+  var match = name.match(/([a-z\-]*):\/\/(.*)/);
+  if (match) {
+    // the http adapter expects the fully qualified name
+    name = /http(s?)/.test(match[1]) ? match[1] + '://' + match[2] : match[2];
+    return {name: name, adapter: match[1]};
+  }
```

Comment:
I removed an "invalid adapter" check here that was already being handled upstream.